gdal/port/cpl_json.h

```diff
@@ -0,0 +1,170 @@
+/******************************************************************************
+ * Project:  Common Portability Library
+ * Purpose:  Function wrapper for libjson-c access.
+ * Author:   Dmitry Baryshnikov, dmitry.baryshnikov@nextgis.com
+ *
+ ******************************************************************************
+ * Copyright (c) 2016-2017 NextGIS, <info@nextgis.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#ifndef CPL_JSON_H_INCLUDED
+#define CPL_JSON_H_INCLUDED
+
+#include "cpl_string.h"
+#include "cpl_progress.h"
+
+/**
+ * \file cpl_json.h
+ *
+ * Interface for read and write JSON documents
+ */
+typedef void *JSONObjectH;
+
+CPL_C_START
+
+class CPLJSONArray;
+/**
+ * @brief The CPLJSONArray class holds JSON object from CPLJSONDocument
+ */
+class CPL_DLL CPLJSONObject
```

Comment:
Did you consider starting from my CPLJsonObject class of frmts/pds/isis3dataset.cpp ? It has some syntaxic suggar using [] operator which makes it similar to Python json handling code

Reply:
Fixed